### PR TITLE
Allow `Field` in `Box`

### PR DESCRIPTION
**Pull request by GuillaumeRx**
This PR allows the usage of `Field` in `Box` to be able to put two field in a row.

### CODE REVIEW

File: packages/snaps-sdk/src/jsx/validation.ts [hunk 1]
```diff
@@ -512,7 +512,7 @@ export const FieldStruct: Describe<FieldElement> = element('Field', {
  */
 export const FormChildStruct = children(
```
Comment (FrederikBolding) [inline packages/snaps-sdk/src/jsx/validation.ts:513]: Can we remove `FormChildStruct` and just use `children(BoxChildStruct)` for `FormStruct`?

Comment (FrederikBolding) [reply packages/snaps-sdk/src/jsx/validation.ts:513]: Or alternatively to not break as much:
`export const FormChildStruct = BoxChildrenStruct`;

**Review by FrederikBolding — APPROVED** [review]

File: packages/snaps-rpc-methods/src/permitted/createInterface.test.tsx [hunk 2]
```diff
@@ -190,7 +191,7 @@ describe('snap_createInterface', () => {
       error: {
         code: -32602,
         message:
-          'Invalid params: At path: ui.props.children -- Expected type to be one of: "Address", "Bold", "Box", "Button", "Copyable", "Divider", "Dropdown", "RadioGroup", "FileInput", "Form", "Heading", "Input", "Image", "Italic", "Link", "Row", "Spinner", "Text", "Tooltip", "Checkbox", "Card", "Icon", "Selector", "Section", "Avatar", but received: "Field".',
+          'Invalid params: At path: ui.props.children.props.children -- Expected the value to satisfy a union of `tuple | tuple | tuple | object | object | object | object | object | object`, but received: [object Object].',
```
Comment (FrederikBolding) [inline packages/snaps-rpc-methods/src/permitted/createInterface.test.tsx:194]: Damn we should really improve the error messaging for Fields

**Review by FrederikBolding — APPROVED** [review]

File: packages/snaps-rpc-methods/src/permitted/createInterface.test.tsx (hunk 2 above)
Comment (GuillaumeRx) [reply packages/snaps-rpc-methods/src/permitted/createInterface.test.tsx:194]: It's so annoying to have to deal with these weird error messages

Comment (FrederikBolding) [reply packages/snaps-rpc-methods/src/permitted/createInterface.test.tsx:194]: https://github.com/MetaMask/snaps/issues/2927

**Pull request merged**